app/src/main/kotlin/compiler/common/diagnostics/Diagnostic.kt

```diff
@@ -38,6 +38,10 @@ sealed class Diagnostic {
         }.toString()
     }
 
+    sealed class ArgumentResolutionErrors() {
```

Comment:
This should extend Diagnostic. Also should't this be after NameResolutionErrors?

Reply:
Also, the name should be singular `ArgumentResolutionError`.

Reply:
I matched it to `NameResolutionErrors`. I think I'll convert this to singular as well.